File: app/src/main/java/com/ipd/taxiu/ui/fragment/group/GroupFragment.java
package com.ipd.taxiu.ui.fragment.group;

import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;

import com.ipd.taxiu.R;
import com.ipd.taxiu.adapter.GroupListAdapter;
import com.ipd.taxiu.adapter.OrderListAdapter;
import com.ipd.taxiu.bean.GroupBean;
import com.ipd.taxiu.bean.GroupListBean;
import com.ipd.taxiu.bean.OrderBean;
import com.ipd.taxiu.bean.OrderListBean;
import com.ipd.taxiu.ui.ListFragment;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;

import rx.Observable;
import rx.Subscriber;

/**
 * Created by Miss on 2018/7/19
 * 拼团
 */
public class GroupFragment extends ListFragment<GroupListBean, GroupBean> {
    private GroupListAdapter mAdapter = null;

    public static GroupFragment newInstance(int categoryId) {
        GroupFragment fragment = new GroupFragment();
        Bundle bundle = new Bundle();
        bundle.putInt("categoryId", categoryId);
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    protected void initView(@Nullable Bundle bundle) {
        super.initView(bundle);
        progress_layout.setEmptyViewRes(R.layout.layout_empty_group);
    }

    @NotNull
    @Override
    public Observable<GroupListBean> loadListData() {
        final Bundle args = getArguments();
        final int categoryId = args.getInt("categoryId");
        return Observable.create(new Observable.OnSubscribe<GroupListBean>() {
            @Override
            public void call(Subscriber<? super GroupListBean> subscriber) {
                GroupListBean bean = new GroupListBean();
                bean.list = new ArrayList<>();
                if (categoryId == 0){
                    bean.list.add(new GroupBean(1));
                }else if (categoryId ==1){
                    bean.list.add(new GroupBean(2));
                }else {
                    bean.list.add(new GroupBean(3));
                }
                subscriber.onNext(bean);
                subscriber.onCompleted();
            }
        });
    }

    @Override
    public int isNoMoreData(GroupListBean result) {
        if (result.list == null || result.list.isEmpty()) {
            return getEMPTY_DATA();
        } else {
            return getNORMAL();
        }
    }

    @Override
    public void setOrNotifyAdapter() {
        if (mAdapter == null) {
            mAdapter = new GroupListAdapter(getContext(), getData());
            recycler_view.setLayoutManager(new LinearLayoutManager(getContext()));
            recycler_view.setAdapter(mAdapter);
        } else {
            mAdapter.notifyDataSetChanged();
        }
    }

    @Override
    public void addData(boolean isRefresh, GroupListBean result) {
        getData().addAll(result.list);
    }
}

File: app/src/main/java/com/ipd/taxiu/ui/fragment/group/GroupFragment.kt
package com.ipd.taxiu.ui.fragment.group

import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import com.ipd.taxiu.R
import com.ipd.taxiu.adapter.GroupListAdapter
import com.ipd.taxiu.bean.BaseResult
import com.ipd.taxiu.bean.GroupBean
import com.ipd.taxiu.platform.global.Constant
import com.ipd.taxiu.platform.global.GlobalParam
import com.ipd.taxiu.platform.http.ApiManager
import com.ipd.taxiu.ui.ListFragment
import com.ipd.taxiu.ui.activity.group.GroupDetailActivity
import rx.Observable

/**
 * Created by Miss on 2018/7/19
 * 拼团
 */
class GroupFragment : ListFragment<BaseResult<List<GroupBean>>, GroupBean>() {
    companion object {
        fun newInstance(categoryId: Int): GroupFragment {
            val fragment = GroupFragment()
            val bundle = Bundle()
            bundle.putInt("categoryId", categoryId)
            fragment.arguments = bundle
            return fragment
        }
    }

    override fun needLazyLoad() = true

    override fun initView(bundle: Bundle?) {
        super.initView(bundle)
        progress_layout.setEmptyViewRes(R.layout.layout_empty_group)
    }

    private val mCategoryId by lazy { arguments.getInt("categoryId", 0) }
    override fun loadListData(): Observable<BaseResult<List<GroupBean>>> {
        return ApiManager.getService().mineSpellList(GlobalParam.getUserId(), page, Constant.PAGE_SIZE, mCategoryId)
    }

    override fun isNoMoreData(result: BaseResult<List<GroupBean>>): Int {
        if (page == INIT_PAGE && (result.data == null || result.data.isEmpty())) {
            return EMPTY_DATA
        } else if (result.data == null || result.data.isEmpty()) {
            return NO_MORE_DATA
        }
        return NORMAL
    }

    private var mAdapter: GroupListAdapter? = null
    override fun setOrNotifyAdapter() {
        if (mAdapter == null) {
            mAdapter = GroupListAdapter(context, data, {
                GroupDetailActivity.launch(mActivity, it.ORDER_ID)
            })
            recycler_view.layoutManager = LinearLayoutManager(context)
            recycler_view.adapter = mAdapter
        } else {
            mAdapter?.notifyDataSetChanged()
        }
    }

    override fun addData(isRefresh: Boolean, result: BaseResult<List<GroupBean>>) {
        data?.addAll(result?.data ?: arrayListOf())
    }

}
